whats-that-typeclass-functor/article.md

```diff
@@ -0,0 +1,467 @@
+You might already know the `map` function for lists:
+
+```haskell
+> map (+1) [1, 2, 3]
```

Comment:
If your audiance does not know what a `functor` is, chances are that they are not really familiar with stuff like `(+1)`, might be even be not familiar with lambdas. I think it is better stick to separate function with explicit type signatures, but here, using a lambda might be a good enough compromise.

Reply:
Oh, that's interesting. 

I think the readers should have enough experience (+ this is kind of intuitive), but a lambda is OK as well. 

On a side note, I believe we can expect around 99% of audience to know what a lambda is or be able to recognize it: it's just so common in 2022.

Reply:
Oh, I have a general preference for keeping everything dead simple, except for the thing that is being explained. I guess I am a bit burned by tutorials that set out to explain something, and use somethings equally obscure as part of the explanation.

For example, back when I wanted to know about data families in Haskell, this is the page I will usually end up at https://wiki.haskell.org/GHC/Type_families, and in that page, at the begining of the explanation, there is something like. 

> As an example, consider Ralf Hinze's [generalised tries](http://www.cs.ox.ac.uk/ralf.hinze/publications/GGTries.ps.gz), a form of generic finite maps.

And I don't have any clue about "tries" let alone "Relf Hinze's generalized" ones, and it seems to me that this aspect makes it only accessable to people coming from some sort of academic background.

So I think if you are explaining something that needs explaining, it is a good idea to keep everything else dead simple, and I think it will considerable expand the size of the audiance that your explanation will work for.

Reply:
Yes, you are entirely correct on that. I try to operate Serokell Blog articles on exactly the same assumption.